File: chartiq/src/main/java/com/chartiq/sdk/model/Study.java
package com.chartiq.sdk.model;

import java.io.Serializable;
import java.util.Map;

/**
 * Created by sobolev on 2/17/17.
 */

public class Study implements Serializable{
    public Map<String, Object> attributes;
    public Double centerline;
    public Boolean customRemoval;
    public Boolean deferUpdate;
    public String display;
    public Map<String, Object> inputs;
    public String name;
    public Map<String, Object> outputs;
    public Boolean overlay;
    public Map<String, Object> parameters;
    public String range;
    public String shortName;
    public Boolean underlay;
    public Map<String, Object> yAxis;
    public String type;

    @Override
    public boolean equals(Object obj) {
        boolean equality = super.equals(obj);
        if(!equality && obj instanceof Study){
            return this.shortName != null && this.shortName.equals(((Study) obj).shortName);
        }
        return equality;
    }
}

File: chartiq/src/main/java/com/chartiq/sdk/model/Study.kt
package com.chartiq.sdk.model

import android.os.Parcelable
import kotlinx.android.parcel.Parcelize
import kotlinx.android.parcel.RawValue

@Parcelize
data class Study(
    val attributes: @RawValue Map<String, Any>?,
    val centerLine: Double,
    val customRemoval: Boolean,
    val deferUpdate: Boolean,
    val display: String?,
    var inputs: @RawValue Map<String, Any>?,
    var outputs: @RawValue Map<String, Any>?,
    val name: String,
    val overlay: Boolean,
    var parameters: @RawValue Map<String, Any>,
    val range: String?,
    val shortName: String,
    val type: String?,
    val underlay: Boolean,
    val yAxis: @RawValue Map<String, Any>?,
) : Parcelable

fun Study.splitName(): Pair<String, String>{
    val result = name.split("\u200C")
    return when (result.size) {
        3 -> Pair(result[1], result[2])
        2 -> Pair(result.first(), result.last())
        else -> Pair(result.toString(), "")
    }
}
